### ISSUE TITLE
Odd behaviour when NVDA is in aragonese

### ISSUE THREAD

**Issue by jmdaweb** [nvaccess/nvda#11155]
### Steps to reproduce:

1. Change NVDA language to aragonese.
2. Open Notepad.
3. Type one or two lines of text.
4. Try navigating by word (control+left or right arrows) or by character (left and right arrows).

### Actual behavior:
NVDA does not read anything.
### Expected behavior:
NVDA should read previous / next character or word, as it does in other languages.
#### NVDA installed/portable/running from source:
Installed and portable.
#### NVDA version:
2020.1
#### Windows version:
Windows 10 version 1909
#### Name and version of other software in use when reproducing the issue:
This issue also affects some system dialogs, Mozilla Firefox and the Windows console with UIA experimental support disabled.
#### Other information about your system:
Relevant log fragment:
```
error executing script: <bound method EditableText.script_caret_moveByWord of <NVDAObjects.Dynamic_IAccessibleEditWindowNVDAObjectEditableIndentNav object at 0x047237D0>> with gesture 'ctrl+flecha dreita'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 205, in executeScript
  File "editableText.pyc", line 226, in script_caret_moveByWord
  File "editableText.pyc", line 164, in _caretMovementScriptHelper
  File "NVDAObjects\behaviors.pyc", line 175, in _caretScriptPostMovedHelper
  File "editableText.pyc", line 149, in _caretScriptPostMovedHelper
  File "textInfos\offsets.pyc", line 506, in expand
  File "textInfos\offsets.pyc", line 485, in _getUnitOffsets
  File "NVDAObjects\window\edit.pyc", line 418, in _getWordOffsets
  File "textInfos\offsets.pyc", line 366, in _getWordOffsets
  File "locale.pyc", line 591, in getlocale
  File "locale.pyc", line 499, in _parse_localename
ValueError: unknown locale: an
```
#### Does the issue still occur after restarting your computer?
Yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2019.2 works as expected. The problem starts in 2019.3.
#### If addons are disabled, is your problem still occuring?
Yes.
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
Yes.

**Comment by josephsl** [comment]
Hi, this is because Python does not recognize Aragonese locale at all, partly caused by the fact that Windows doesn’t recognize Aragonese locale properly. This isn’t limited to Aragonese either: Central and Northern Kurdish and a handful of other languages are also affected. Although Windows 8.x and 10 recognizes more languages and locales, these releases do not recognize some of them that NVDA has been translated into. Thanks.

**Comment by lukaszgo1** [comment]
cc @leonardder as you've written this part of the code for Python 3. Looking at it using code page returned by locale.getlocale seems quite strange to me. My understanding is that we want the Windows Ansi code page here not the code page of the NVDA locale in use. If so I propose to replace these calls to locale.getlocale with locale.getpreferredencoding.

**Comment by leonardder** [comment]
@lukaszgo1 Could you please elaborate on what you mean with `these calls`? Something like a file/line number would be helpful.

**Comment by lukaszgo1** [comment]
I mean lines:
285, 349, 366 and  400 of source\textInfos\offsets.py

**Comment by ivnc** [comment]
Are there nay news on this? The Aragonese translator has complanied again through the Spanish-speaking list about this issue.

Regards.

**Comment by lukaszgo1** [comment]
@leonardder Have you seen my commend above? Could you explain why you've chosen to use locale.getlocale rather than locale.getpreferredencoding?

**Comment by lukaszgo1** [comment]
@leonardder  Ping

**Comment by lukaszgo1** [comment]
@jmdaweb @ivnc Could you please test with [this build](https://ci.appveyor.com/api/buildjobs/m2sc98go0i3aomie/artifacts/output%2Fnvda_snapshot_pr11384-20547%2Cfa7dddac.exe),  perhaps pass it to the Aragonese translator if that  is not too much trouble, and confirm that the issue is fixed for you?

**Comment by jmdaweb** [comment]
Hi,
The aragonese translator confirms the custom build works as expected.
Regards.
